server/api/src/rsDataObjOpen.cpp

```diff
@@ -526,6 +602,39 @@ int change_replica_status_to_intermediate(
     return status;
 } // change_replica_status_to_intermediate
 
+int change_replica_status(rsComm_t& rsComm, dataObjInp_t& dataObjInp, int new_replica_status)
+{
+    {
+        const auto l = {GOOD_REPLICA, INTERMEDIATE_REPLICA, STALE_REPLICA};
+        if (std::none_of(std::begin(l), std::end(l), [x = new_replica_status](auto y) { return x == y; })) {
+            return SYS_INVALID_INPUT_PARAM;
+        }
+    }
+
+    dataObjInfo_t info{};
+    rstrcpy(info.objPath, dataObjInp.objPath, MAX_NAME_LEN);
+
+    keyValPair_t kvp{};
+    irods::experimental::key_value_proxy dst{kvp};
+    irods::experimental::key_value_proxy src{dataObjInp.condInput};
+
+    dst[RESC_HIER_STR_KW] = src[RESC_HIER_STR_KW];
```

Comment:
the resource hierarchy is supposed to be set in the dataObjInfo_t, not the keyValPair_t.